Module_02/README.md

```diff
@@ -1,5 +1,307 @@
+
+
+
+**Further Reading on Version Control and Git**
+
+https://www.atlassian.com/git/tutorials/what-is-version-control
+
+https://www.atlassian.com/git/tutorials/what-is-git
+
+
+
+### Git vs Github
+
+Git is a version control software.
+
+Github is a cloud storage service for Git repositories.
+
+
+
+### Connecting Your Github Account to Your Local Installation of Git and Setting a Default Editor
+
+1) You'll run the following commands to edit your local git configuration. You'll use youre Github handle and email. Here I've used mine as an example:
+
+```bash
+git config --global user.name "zan-rosenthal"
+git config --global user.email zan.rosenthal@gmail.com
+```
+
+2) Now, set your default editor for git. This will allow you to write messages in your git history:
+
+```bash
+git config --global core.editor "code --wait"
+```
+
+
+
+###Starting a Repo
+
+A repo, or repository is an object that contains all versions of a specific codebase. Going back to the Google Docs model, the repository is like a folder where all the different copies of the a single document live. However, instead of keeping every copy of the codebase on your computer, Git manages snapshots or commits. You can use git commands to check out any commit in your git history, which will instantly change the code you have on your computer to look like it did at the point of that commit. 
+
+
+
+1)In the terminal, navigate to the top level directory of one of the projects you've been working on in a previous modules
+
+2) Then type the following command:
+
+```bash
+git init
+```
+
+This will create a git repo in your current directory. Now we can start tracking changes.
+
+
+
+**Making Your First Commit**
+
+3) Make a change in your project. Next run:
+
+```bash
+git status
+```
+
+This will show the names for the files that have been modified. Git is watching all changes in this directory. 
+
+4) Try:
+
+```bash
+git diff
+```
+
+This will show the contents of the file that have been changed and outline the delta, or changes, in the file. 
+
+5) Now run:
+
+```bash
+git add [path to file]
+git status
+```
+
+You'll noticed the color of the file name has changed. By running `git add` you've staged the files for commit. This means when we make a commit, those staged files will be included in the snapshot represented by that commit. Now we'll commit them and add the current snapshot of the directory to our git history. 
+
+6) Run:
+
+```bash
+git commit
+```
+
+This should open your editor to allow you write a message describing the commit. Git commit messages should be short and descriptive and should use action verbs. For instance:
```

Comment:
This defaults to a vim editor right? should we tell them `:wq` to write and quit?

Reply:
In an earlier section, I had them set their default git editor to vs code.